### PR TITLE
Change incorrect bit masking in BME680 XLSBs

### PR DESCRIPTION
**Pull request by JamesTimothyMeech**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,7 +135,7 @@ Select:
 - 'u': set I2C address.
 - 'x': disable SWD and spin for 10 secs.
 - 'z': dump all sensors data.
-Enter selection>
+Enter selection>
```
Comment (phillipstanleymarbell) [inline README.md:138]: This change is not relevant to the pull request.

File: src/boot/ksdk1.1.0/devBME680.c [hunk 2]
```diff
@@ -215,7 +215,7 @@ printSensorDataBME680(bool hexModeFlag, uint16_t menuI2cPullupValue)
 	unsignedRawAdcValue =
 			((readSensorRegisterValueMSB & 0xFF)  << 12) |
 			((readSensorRegisterValueLSB & 0xFF)  << 4)  |
-			((readSensorRegisterValueXLSB & 0xF0) >> 4);
+			((readSensorRegisterValueXLSB & 0x0F) >> 4);
```
Comment (phillipstanleymarbell) [inline src/boot/ksdk1.1.0/devBME680.c:218]: Can you explain why you think it is meaningful to mask off the the upper bits keeping only the lower four bits by `(readSensorRegisterValueXLSB & 0x0F)` and then shifting right by four to throw away the bits you just purposefully kept? Double-check your reasoning (see **Section 5.3.4.2 Temp data** of the BME680 manual).

**Comment by JamesTimothyMeech** [comment]
The oversampling controls the resolution. See page 17 of the datasheet Section 3.3.1: 
"When the IIR filter is disabled, the temperature resolution is 16 + (osrs_t-1) bit, e.g. 18 bit when osrs_t is set to '3'"

**Comment by phillipstanleymarbell** [comment]
> The oversampling controls the resolution. See page 17 of the datasheet Section 3.3.1:
> "When the IIR filter is disabled, the temperature resolution is 16 + (osrs_t-1) bit, e.g. 18 bit when osrs_t is set to '3'"

Yes, but you're confusing the number of bits you get back with the effective resolution in those bits.

**Comment by JamesTimothyMeech** [comment]
Section 5.3.4.2 Temp data says "Contains the XLSB part [3:0] of the raw
temperature measurement output data. Contents
depend on temperature resolution controlled by
oversampling setting. " this makes it sound like oversampling leads to bits appearing in the LSB. What do you think ?

**Comment by phillipstanleymarbell** [comment]
@JamesTimothyMeech, should we keep this open or do you want to close it?

**Comment by JamesTimothyMeech** [comment]
Setting oversampling to n causes the sensor to take n readings, average them and store the average in the corresponding MSB and LSB registers. Based on the information above I believe that the sensor is designed to return 0 for the XLSB until oversampling is enabled. This is because without oversampling the digits in the XLSB will be down in the noise and only show random variation. Oversampling will hide the random variation through averaging, after a certain amount of oversampling the XLSB value will appear stable due to the averaging. If we enable oversampling the XLSB will return some value other than zero. Could we merge the pull request and collect some data to test this hypothesis?

**Comment by phillipstanleymarbell** [comment]
Here is an example output with the change:
```
59, 1451692823, 24424, 0x00 0x59, 0x00 0x5A, 0x00 0x59, 0x00 0x5B, 0x00 0x5B, 0x00 0x58, 0x00 0x55, 0x00 0x4D, 0x00 0x4D, 0x00 0x4E, 0x00 0x4E, 0x00 0x4E, 0x00 0x4D, 0x00 0x4B, 0x00 0x49, 0x00 0x40, 0x00 0x49, 0x00 0x45, 0x00 0x45, 0x00 0x46, 0x00 0x46, 0x00 0x43, 0x00 0x43, 0x00 0x39, 0x00 0x42, 0x00 0x43, 0x00 0x40, 0x00 0x40, 0x00 0x40, 0x00 0x3E, 0x00 0x3D, 0x00 0x34, 0x00 0x3C, 0x00 0x3C, 0x00 0x3B, 0x00 0x3D, 0x00 0x3D, 0x00 0x39, 0x00 0x38, 0x00 0x2D, 0x00 0x39, 0x00 0x36, 0x00 0x38, 0x00 0x38, 0x00 0x38, 0x00 0x33, 0x00 0x31, 0x00 0x23, 0x00 0x2B, 0x00 0x2D, 0x00 0x34, 0x00 0x31, 0x00 0x2F, 0x00 0x30, 0x00 0x27, 0x00 0x19, 0x00 0x19, 0x00 0x22, 0x00 0x2A, 0x00 0x28, 0x00 0x25, 0x00 0x26, 0x00 0x1C, 0x0F 0xFF, 0x02 0xB2, 0xFE 0xF0, 0xFC 0xE0, 0xC0 0x00, 0x05 0xF0, 0xFF 0x48, 0xF8 0x1B, 0x14, 0xFF 0x8B, 0xFF 0x55, 0x00 0x36, 0xFD, 0x55 0xBD 0x00, 0x80 0x8F 0x00, 0x46 0x3B, ----, ----, ----, ----, ----, ----, ----, ----, 0x00 0x01, 0x00 0x02, 0x00 0x00, ----, ----, ----, ----, ----, 0x70 0x44, 0x3A 0xA4, 1451692823, 26115, 8
```

**Comment by JamesTimothyMeech** [comment]
I see, I was incorrect. I'll take another look at this

**Comment by phillipstanleymarbell** [comment]
OK, I'll revert the merge

**Pull request merged**